Bio/SeqFeature.py

```diff
@@ -1188,6 +1198,24 @@ def __iter__(self):
             for pos in loc:
                 yield pos
 
+    def __eq__(self, other):
+        """Check if all parts of CompoundLocation are equal to all parts of other CompoundLocation"""
+        if not isinstance(other, CompoundLocation):
+            return False
+        if not len(self.parts) == len(other.parts):
+            return False
+        if not self.operator == other.operator:
+            return False
+        identical = True
+        for i, self_part in enumerate(self.parts):
```

Comment:
Would ``for self_part, other_part in zip(self.parts, other.parts):`` be nicer here given you've confirmed the have the same number of parts above? e.g.

```python
for self_part, other_part in zip(self.parts, other.parts):
    if not self_part == other_part:
        return False
return True
```